pkg/api/v1/types.go

```diff
@@ -1051,15 +1051,22 @@ type ClusterStatus struct {
 	CCM ExternalCCMStatus `json:"ccm"`
 }
 
+type ExternalCCMMigrationStatus string
+
+var (
+	ExternalCCMMigrationNotNeeded   ExternalCCMMigrationStatus = "notNeeded"
+	ExternalCCMMigrationSupported   ExternalCCMMigrationStatus = "supported"
+	ExternalCCMMigrationUnsupported ExternalCCMMigrationStatus = "unsupported"
+	ExternalCCMMigrationInProgress  ExternalCCMMigrationStatus = "inProgress"
```

Comment:
Tiny nit: usually status strings start with an uppercase letter. I don't think it's important, but I'll leave it up to @maciaszczykm to decide should this be changed or not.

Reply:
I agree, let's align it with other places.